build.js

```diff
@@ -1,59 +1,43 @@
-const fs = require("fs").promises;
-const path = require("path");
-
-const sourceFiles = [
-  "src/catppuccin-frappe.theme.scss",
-  "src/catppuccin-latte.theme.scss",
-  "src/catppuccin-macchiato.theme.scss",
-  "src/catppuccin-mocha.theme.scss",
-];
-
-const accents = [
-  "rosewater",
-  "flamingo",
-  "pink",
-  "mauve",
-  "red",
-  "maroon",
-  "peach",
-  "yellow",
-  "green",
-  "teal",
-  "sky",
-  "sapphire",
-  "blue",
-  "lavender",
-];
-
-(async () => {
-  await Promise.all(sourceFiles.map(generateAccents));
-  console.log("Generated all accents for all flavours");
-})();
-
-// read sourceFile and generate all accents for it
-async function generateAccents(sourceFilePath) {
-  const _sourceFilePath = path.join(__dirname, sourceFilePath);
-  const sourceFileData = await fs.readFile(_sourceFilePath, {
-    encoding: "utf8",
-  });
-  return Promise.all(
-    accents.map((accent) =>
-      generateAccent(sourceFileData, sourceFilePath, accent)
-    )
-  );
-}
-
-// replace brand and write to separate file
-async function generateAccent(sourceFileData, sourceFilePath, accent) {
-  const modifiedFileContent = sourceFileData.replace(
-    /\$brand: .*;/gm,
-    `$brand: \$${accent};`
-  );
-  const outputFileName = sourceFilePath
-    .split(".")
-    .map((s, i) => (i === 0 ? s.concat(`-${accent}`) : s))
-    .join(".");
-  const outputFilePath = path.join(__dirname, outputFileName);
-  await fs.writeFile(outputFilePath, modifiedFileContent);
-  console.log(`Generated: ${outputFileName}`);
-}
+import fs from "fs/promises";
+import path from "path";
+import { initAsyncCompiler } from "sass-embedded";
+
+import { flavorEntries } from "@catppuccin/palette";
+
+const DEFAULT_ACCENT = "blue";
+
+await fs.rm("dist/", { recursive: true, force: true });
+await fs.mkdir("dist/dist/", { recursive: true });
+
+const compiler = await initAsyncCompiler();
+
+await Promise.all(
+  flavorEntries.map(async ([flavor, { colorEntries }]) => {
+    const src = `src/catppuccin-${flavor}.theme.scss`;
+    const contents = await fs.readFile(src, "utf-8");
+
+    await Promise.all(
+      colorEntries.map(async ([accent, { hex, accent: isAccent }]) => {
+        if (!isAccent) return;
+        const { css } = await compiler.compileStringAsync(
+          `$brand: ${hex};\n` + contents,
+          {
+            style: "compressed",
+            loadPaths: ["node_modules/", "src/"],
+            silenceDeprecations: ["color-functions"],
```

Comment:
This is needed since we are currently using deprecated Sass functions to adjust lightness of colors.

```
Deprecation Warning: lighten() is deprecated. Suggestions:

color.scale($color, $lightness: 100%)
color.adjust($color, $lightness: 35%)

More info: https://sass-lang.com/d/color-functions

  ╷
9 │   --brand-experiment-230: #{lighten($brand, math.div(70%, 2))};
  │                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
    ../../../../src/components/_variables.scss 9:29  @import
    _theme.scss 1:9                                  @import
    - 16:9                                           root stylesheet
```